pico/configdriven/runtime/src/main/java/io/helidon/pico/configdriven/runtime/ConfigDrivenServiceProviderBase.java

```diff
@@ -712,32 +571,47 @@ void resolveConfigDrivenServices() {
         }
 
         // accept and resolve config
-        managedConfiguredServicesMap.values().forEach(opt -> {
-            assert (opt.isPresent());
-
-            AbstractConfiguredServiceProvider<?, CB> csp = opt.get();
+        managedConfiguredServicesMap.values().forEach(csp -> {
             if (LOGGER.isLoggable(System.Logger.Level.DEBUG)) {
                 LOGGER.log(System.Logger.Level.DEBUG, "Resolving config for " + csp);
             }
 
             LogEntryAndResult logEntryAndResult = createLogEntryAndResult(Phase.PENDING);
             try {
-                if (csp.configBean().isEmpty()) {
```

Comment:
why was this removed?

Reply:
Some states are now impossible. I have changed the initialization to be a one-pass thing, that creates service instances with the config beans. There is a single set of config beans that are created at a single point in time, and these are then used to drive instantiation of services. The config bean cannot be empty on a non-root service provider (as it is a final field), and when you try to get it from root, you get an exception.